fc_util/src/time.rs

```diff
@@ -0,0 +1,209 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

Comment:
I understand why you want to have your own time implementation for at least some operations here for truly monotonic clocks, but it feels like some of the operations here could be refactored further to use std::time::Duration since the current API is kind of messy right now and std::time::SystemTime/Instant could handle some of the other operations, for example for logging purposes.

Reply:
Your suggestion are valid and worth looking at however this is out of the scope of this PR. If you have suggestion of improvements for our time module from the `fc_util` crate, I encourage you to open an issue where we can take a deeper look at what these changes may involve.